owiddata/generate-owiddata-stats.py

```diff
@@ -45,6 +73,52 @@ def convert_date(git_date):
     # Assumes the year will not begin with 0
     return datetime.datetime.fromisoformat(git_date).strftime('%B %d, %Y').replace(' 0', ' ')
 
+def retrieve_platform_types():
+    """Use trackvaccines.org to scrape the website listing approved vaccines
+    Returns: dataframe """
+    vaccine_info = dict()
+    vaccineHTML = urllib.request.urlopen('https://covid19.trackvaccines.org/vaccines/approved/')
+
+    # Extract the HTML that makes the cards on the webpage (each card is a vax)
+    soup = BeautifulSoup(vaccineHTML, "html5lib")
+    body = soup.find('body')
+    cards = body.find_all('li')
+
+    # Iterate through the cards to extract information
+    # Tags were identified empirically and are not self-evident
+    for card in cards: # find all element of tag
+        if card.find('a', {"class": "icon-link"}) is not None:
+            vaccine_type = card.find('a', {"class": "icon-link"}).get_text()
+            if vaccine_type.upper() != vaccine_type: #DNA, RNA, VLP
+                vaccine_type = vaccine_type.lower()
+            vaccine_category = platforms(vaccine_type)
+            vaccine_manf = card.find('span',
+                                     {"class": "has-medium-font-size"}).get_text()
+            vaccine_name = card.find('span',
+                                     {"class": "has-large-font-size"}).get_text()
+            link = card.find('a', href=True)
+            vaccine_info[vaccine_name] = [vaccine_manf,
+                                          vaccine_type,
+                                          vaccine_category,
+                                          link['href']]
+    vaccine_df = pd.DataFrame.from_dict(vaccine_info, orient='index')
+    vaccine_df.rename(mapper={0: "Company", 1: "Type", 2: "Category", 3: "URL"},
+                      axis=1, inplace=True)
+    vaccine_df.index.name = 'Vaccine'
+    vaccine_df["Type"] = vaccine_df["Type"].replace("DNA","plasmid vectored")
+    return vaccine_df
+
+def create_table(vaccine_df, category):
+    """For each vaccine type, select a subset of the vaccine information table
+    Input: dataframe, string
+    Returns: string representing a table in markdown"""
+    vaccines = vaccine_df[vaccine_df["Category"] == category]
+    numTypes = len(set(vaccines["Type"].to_list()))
+    if numTypes > 1:
+        return vaccines[["Company", "Type"]].to_markdown()
```

Comment:
I'm not sure how the JSON will store the markdown.  We can see what happens and adjust if needed.

Reply:
I'm curious if this will work!!! In theory I think the JSON looked ok but there are some special characters that could make things interesting